pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -272,20 +279,17 @@ cdef class StringVector(Vector):
         if not self.data.data:
             raise MemoryError()
 
-    cdef resize(self):
+    cdef resize(self, size_t new_size):
         cdef:
             char **orig_data
             Py_ssize_t i, m
 
         m = self.data.m
-        self.data.m = max(self.data.m * 4, _INIT_VEC_CAP)
+        self.data.m = max(new_size, _INIT_VEC_CAP)
 
-        orig_data = self.data.data
-        self.data.data = <char **>malloc(self.data.m * sizeof(char *))
+        self.data.data = <char **>realloc(self.data, self.data.m * sizeof(char *))
         if not self.data.data:
             raise MemoryError()
-        for i in range(m):
```

Comment:
There are still a few other places where we are doing loop assignment in the object class, but I have not touched them because I don't know what Cython does for reference counting during assignment that a simple mempcy would not know about. Something to research futher